File: src/Tbd.WebApi.Shared/Extensions/ApiVersioningMiddlewareExtensions.cs
namespace Tbd.WebApi.Shared.Extensions;

public static class ApiVersioningMiddlewareExtensions
{
    public static void AddApiVersioningEx(this IServiceCollection services)
    {
        services.AddApiVersioning(options =>
        {
            options.ReportApiVersions = true;
            options.AssumeDefaultVersionWhenUnspecified = true;
            options.DefaultApiVersion = new ApiVersion(1, 0);
        });
    }
}

File: src/RecipeApp.FCoreApi/Features/Extensions/ApiVersioningMiddlewareExtensions.fs
namespace RecipeApp.FCoreApi.Features.Extensions

open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.DependencyInjection

module ApiVersioningMiddlewareExtensions =
    type IServiceCollection with
        member this.AddApiVersioningEx() =
            this.AddApiVersioning(fun options ->
                options.ReportApiVersions <- true
                options.AssumeDefaultVersionWhenUnspecified <- true
                options.DefaultApiVersion <- new ApiVersion(majorVersion = 1, minorVersion = 0))
            |> ignore
            this
